src/care_communication/api/auth/kiosk_auth.py

```
@@ -1,37 +1,47 @@
-        except Patient.DoesNotExist:
-            raise AuthenticationFailed("User is not authorized to access patient data")
+        if not (birth_year or phone_number):
+            raise AuthenticationFailed("Either birth year or phone number is required")
 
         try:
-            birth_year = int(birth_year)
-        except ValueError:
-            raise AuthenticationFailed("Invalid birth year")
+            encounter = Encounter.objects.select_related("patient").get(external_id=encounter_id)
+        except Encounter.DoesNotExist:
+            raise AuthenticationFailed("Invalid encounter ID.")
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)